File: 2025/Day1/Solution.py
current=50
answer=0
with open("2025\Day1\input.txt", "r") as file:
    data = file.readline()
    while data:
        direction, value = data[0], int(data[1:])
        # print(direction, value)
        if direction == "L":
            current -= value    
        else:
            current += value
        current%=100
        if current == 0:
            answer +=1
        data = file.readline()
print("First Answer:",answer)

current=50
answer=0
with open("2025\Day1\input.txt", "r") as file:
    data = file.readline()
    while data:
        direction, value = data[0], int(data[1:])
        # print(direction, value)
        answer+=value//100
        value%=100
        for i in range(0, value):
            if direction == "L":
                current-=1
            else:
                current +=1
            current%=100
            if current==0:
                answer+=1
        current%=100
        data = file.readline()
print("Second Answer:",answer)


current = 50
answer = 0

with open("2025\Day1\input.txt", "r") as file:
    data = file.readline()
    while data:
        direction, value = data[0], int(data[1:])
        answer += value // 100
        value %= 100
        if direction == "L":
            if current !=0 and (current - value) <= 0:
                answer += 1
            current -= value
        else:
            current += value
            if current >= 100:
                answer += 1
        current %= 100
        
        data = file.readline()

print("Second Answer(Updated):",answer)

File: 2025/Day1/Solution.js
const fs = require('fs');
const path = "2025/Day1/input.txt"; // Path to your input file

let answer1 = 0;
let answer2 = 0;
let current = 50;

try {
    const data = fs.readFileSync(path, 'utf8');
    const lines = data.split(/\r?\n/);

    for (const line of lines) {

        const direction = line[0];
        let value = parseInt(line.slice(1), 10);
        answer2 += Math.floor(value / 100);
        value = value % 100;

        if (direction === 'L') {
            if (current !== 0 && (current - value) <= 0) {
                answer2 += 1;
            }
            current -= value;
        } else if (direction === 'R') {
            current += value;
            if (current >= 100) {
                answer2 += 1;
            }
        }
        current = current % 100;
        if (current < 0) current += 100;
        if (current === 0) {
            answer1 += 1;
        }
    }

    console.log("First Answer:", answer1);
    console.log("Second Answer:", answer2);

} catch (err) {
    console.error("Error reading file:", err.message);
}
